clarify possible instances of zeebe rollback (#4064)
camunda/camunda-docs: docs/self-managed/zeebe-deployment/operations/update-zeebe.md

This is not permitted because pre-release versions such as alpha releases are considered unstable and do not guarantee compatibility with any other version.
If you attempted to update from a normal version to a pre-release version, you can roll back to the previous normal version.

:::note

> While in most instances version rollbacks are not supported, in this particular use case a version rollback is possible.

Reviewer: ```suggestion
While in most instances version rollbacks are not supported, in this particular use case a version rollback of Zeebe is possible because the "new" version never became active, thanks to the safety check.
```

Reviewer 2: I'm not sure the extra explanation adds a lot here. We don't otherwise talk about what the safety checks or doing, and can possibly introduce new points of confusion re: "new" vs "old" instead of clarifying. If we really think our developers need this info, I'd suggest something like: 

`..a version rollback of Zeebe is possible, as the included safety checks prevented the pre-release version from becoming active.`

Reviewer: I'd keep the addition of "Zeebe" then because rollback of the other applications might not be possible.

Author: Updated in latest commit -- let me know what you think.

> While in most instances version rollbacks are not supported, in this particular use case a version rollback is possible.

Reviewer 2: Could we condense these two lines? Either as a single note, or as one line in the `Cannot upgrade to or from a pre-release version` paragraph, to make it clear this is referring to that error only. 

`If you attempted to update from a minor release to a pre-release or alpha version, it is possible to roll back to the previous version of Zeebe. Note that version rollbacks are not supported in most other instances.`

Author: Updated in latest commit.

If this persists, you may want to [force the update](#rolling-update-is-not-completing).

:::note

> In this specific case, a rollback is possible because users may restart Zeebe with the "skipped" minor version. [Backup and restore](/self-managed/operational-guides/backup-restore/backup-and-restore.md) is possible, but results in partial data loss of everything following the backup. Therefore, a rollback is preferable in this use case.

Reviewer: :thought_balloon: I think including backup and restore here adds to the confusion, especially because _taking_ a backup is not possible in this state. Restoring from a backup is _always_ possible, no need to mention it here.

Author: Updated in latest commit 👍

> 
> If this persists, you may want to [force the update](#rolling-update-is-not-completing).
> 
> :::note
> In this specific case, a rollback is possible because users may restart Zeebe with the "skipped" minor version. [Backup and restore](/self-managed/operational-guides/backup-restore/backup-and-restore.md) is possible, but results in partial data loss of everything following the backup. Therefore, a rollback is preferable in this use case.

Reviewer 2: Similar to above, is it beneficial to combine this section?
```suggestion

If this persists, you can [force the update](#rolling-update-is-not-completing). Alternatively, it is possible to restart Zeebe with the "skipped" minor version.  Note that version rollbacks are not supported in most other instances.
```

Author: Adjusted in latest commit.
